101-VM-Virus-Attack-Prevention/azuredeploy.json

```diff
@@ -0,0 +1,418 @@
+        }
+      },
+      "_artifactsLocationSasToken": {
+        "type": "securestring",
+        "defaultValue": "",
+        "metadata": {
+          "description": "this will be the sas key to access artifacts"
+        }
+      },
+      "location": {
+        "type": "string",
+        "allowedValues": [
+          "East US",
+          "West Europe",
+          "Southeast Asia",
+          "Australia Southeast"
+        ],
+        "defaultValue": "[resourceGroup().location]",
+        "metadata": {
+          "description": "your resources will be create in this location"
+        }
+      },
+      "omsSku": {
+        "type": "string",
+        "defaultValue": "free",
+        "allowedValues": [
+          "Free",
+          "Standalone",
+          "PerNode"
+        ],
+        "metadata": {
+          "description": "this will be you SKU for OMS"
+        }
+      },
+      "vNetName": {
+        "type": "string",
+        "defaultValue": "Virus-Attack-Vnet",
+        "metadata": {
+          "description": "this will be name of the VNET"
+        }
+      },
+      "vNetAddressSpace": {
+        "type": "string",
+        "defaultValue": "10.1.0.0/16",
+        "metadata": {
+          "description": "this will be address space for VNET"
+        }
+      },
+      "subnetName": {
+        "type": "string",
+        "defaultValue": "Subnet-01",
+        "metadata": {
+          "description": "this will be name of the subnet"
+        }
+      },
+      "subnetAddressSpace": {
+        "type": "string",
+        "defaultValue": "10.1.0.0/24",
+        "metadata": {
+          "description": "this will be address space for subnet"
+        }
+      },
+      "nsgNamePrefix": {
+        "type": "string",
+        "defaultValue": "virus-attack",
+        "metadata": {
+          "description": "this will be prefix used for NSG"
+        }
+      },
+      "pipAddressType": {
+        "type": "string",
+        "defaultValue": "dynamic",
+        "allowedValues": [
+          "dynamic",
+          "static"
+        ],
+        "metadata": {
+          "description": "this will be the type of public IP address used for the VM name"
+        }
+      },
+      "vmNamePrefix": {
+        "type": "string",
+        "defaultValue": "attacked-vm",
+        "metadata": {
+          "description": "this will be the prefix used for the VM name"
+        }
+      },
+      "adminUserName": {
+        "type": "string",
+        "defaultValue": "VmUser",
+        "metadata": {
+          "description": "this will be the user name of the VMs deployed"
+        }
+      },
+      "adminUserPassword": {
+        "type": "securestring",
+        "defaultValue": "Password@12345",
+        "metadata": {
+          "description": "this will be the password of the user created on the user"
+        }
+      }
+    },
+    "variables": {
+      "omsWorkspaceName": "[concat('Virus-Attack','-',uniqueString(subscription().id))]",
```

Comment:
this will only generate a unique value within the subscription, which mean only one deployment at a time can be done?

Reply:
yes, good point !
let me make it unique for just the resource group.